build/teamcity-support.sh

```diff
@@ -24,17 +24,54 @@ run() {
   "$@"
 }
 
+# Returns the list of release branches from origin (origin/release-*), ordered
+# by version (bigger version numbers first).
+get_release_branches() {
+  git branch -r --format='%(refname)' \
+    | sed 's/^refs\/remotes\///' \
+    | grep 'origin\/release-*' \
+    | sort -t- -k2 -g -r
+}
+
+# Returns the number of commits in the curent branch that are not shared with
+# the given branch.
+get_branch_distance() {
+  git rev-list --count $1..HEAD
+}
+
+# Returns the closest branch between origin/master and origin/release-* branches.
```

Comment:
this will cause an odd effect if we ever have a situation where the release branch was just forked, and gets some hot fixes before the `master` branch is worked on further.

I think it's fine (since it won't affect us right away) but I think it should be explained / called out in the release process doc, "if the release process wrt branches ever changes, don't forget about updating that code".